tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,506 @@
+
+Hspec-golden provides us with a `defaultGolden` function which creates the golden files and compares it with the Subject Under Test (SUT) output. It takes two parameters:
+
+  - the **name** of the test 
+  - the output of our SUT 
+
+**Note** : _It is important to give the tests unique names otherwise `hspec-golden` won't work_.
+
+```haskell
+defaultGolden :: String -> String -> Golden String
+```
+
+By default this function searches for golden tests inside a `.golden/` directory. This directory is created automatically when the test runs.
+
+Okay, enough reading, let's create our tests. Let's first create a test module named `FizzBuzzGoldenSpec` under `test/FizzBuzz/` directory.
+
+```haskell
+module FizzBuzz.FizzBuzzGoldenSpec where
+```
+
+We need some imports.
+
+```haskell
+import           Test.Hspec                   --Tests
+import           Test.Hspec.Golden            --Golden Tests
+import           FIZZBUZ.FizzBuzz             -- SUT
+```
+Finally let's write the test.
+
+```haskell
+spec :: Spec
+spec = 
+    describe "fizzBuzz" $
+      it "Turns 3 multiples to fizz and 5 multiples to buzz" $
+        defaultGolden "fizzbuzz" (show $ fizzBuzz [1,2,3,4,5,11,12,13,14,15])
+```
+We are now ready to test our module.
+
+```shell
+$ stack test
+
+hspec-golden-tests> test (suite: hspec-golden-tests-test)
+
+
+FizzBuzz.FizzBuzzGolden
+  fizzBuzz
+    Turns 3 multiples to fizz and 5 multiples to buzz
+      First time execution. Golden file created.
+
+Hello.Hello
+  sayHi
+    shows a Hello Golden Testers string
+
+Finished in 0.0006 seconds
+2 examples, 0 failures
+
+hspec-golden-tests> Test suite hspec-golden-tests-test passed
+```
+
+The test is successful. If we check the ".golden" directory we will find two files.
+
+ - golden : Stores the expected output.
+ - actual : Stores the real output from the SUT.
+
+
+
+```shell
+$ tree
+.golden
+ └── fizzbuzz
+     ├── actual
+     └── golden
+```
+
+The testing framework recognized that this was the first execution, therefore created the fizzbuzz test with the `actual` and `golden` files. The difference between these two files is that the `golden` file will stay the same unless we want to update it while the `actual` file will be overwritten every time we run the test. This is useful when updating the tests, but we will see that later on in this tutorial. 
+
+
+## A More Real Test Case
+
+The `fizzbuzz` module was used to create a simple and didactic example but it could have been easily tested with unit tests. In practice golden tests are often used when testing JSON, HTML or images which generate large outputs. We also haven't yet tested the entire functionality of the `hspec-golden` library yet. For example the `hgold` CLI for updating the golden files.
```

Comment:
Try to avoid passive voice like "was used to create" or "is often used" whenever possible. I think it's better to use "we used whatever to create whatever" or "We used this thing to do that thing"